### DISCUSSION TITLE
An idea to create a strong circumvent tool

### DISCUSSION THREAD

**Discussion by Arya2010** [GFW-knocker/gfw_resist_tls_proxy#95] [Ideas]
I was wondering if we could combine Geneva project codes with any proxy code to create a strong circumvent tool which will be able to detect the DPI censor filtering type and encrypt the data in parallel. 
Because Geneva is not a safe circumvention tool by its own (like proxies and VPNs). Its just an AI strategy to bypass TLS blocking. 
 
Here is the project address: 
https://github.com/Kkevsterrr/geneva

**Comment by GFW-knocker** [comment]
geneva is powerful in low-level network manipulation.
thank you for mention it

**Reply by Arya2010** [reply]
Based on the reports published on their websites, it could not be a low-level network manipulation tool. Because they have discovered many strategies facing the GFW of China , India and Iran and they have helped  tool's programmers(Outline & V2ray) to change the strategy against the firewall. 
So seems that this is the best tool to detect the DPI type right now instead of guessing how DPI works on Irancell and MCI (Hamrah-e-Aval)
